include/dlaf/util_matrix.h

```diff
@@ -137,17 +137,17 @@ class getter_random<std::complex<T>> : private getter_random<T> {
 }
 
 /// Sets all the elements of all the tiles to zero
-template <class T, class ExecutorOrPolicy>
-void set0(ExecutorOrPolicy ex, Matrix<T, Device::CPU>& matrix) {
+template <class T, Device D, class ExecutorOrPolicy>
+void set0(ExecutorOrPolicy ex, Matrix<T, D>& matrix) {
   for (const auto& idx : iterate_range2d(matrix.distribution().localNrTiles()))
-    matrix(idx).then(ex, hpx::unwrapping(tile::set0<T>));
+    matrix(idx).then(ex, matrix::unwrapExtendTiles(tile::set0_o));
```

Comment:
note to self: use `dataflow` instead of `.then` because of the (current) limitation of the CUDA executor